contrib/automation_tests/test_cases/capture_window.py

```diff
@@ -338,3 +345,55 @@ def _execute(self, track_name_filter: str, expect_exists: bool = True):
                 self.expect_true(track.callstacks is not None, 'Track "%s" has callstacks pane' % track.name)
             else:
                 self.expect_true(track.callstacks is None, 'Track "%s" has no callstacks pane' % track.name)
+
+
+class SetAndCheckMemorySamplingPeriod(E2ETestCase):
+
+  def _show_capture_options_dialog(self):
+      logging.info('Opening "Capture Options" dialog')
+      capture_tab = self.find_control('Group', 'CaptureTab')
+      capture_options_button = self.find_control('Button', 'Capture Options', parent=capture_tab)
+      capture_options_button.click_input()
+
+  def _enable_collect_system_memory_usage(self):
+      logging.info('Selecting "Collect system-wide memory usage information" checkbox')
+      collect_system_memory_usage_checkbox = self.find_control(
+          'CheckBox', 'Collect system-wide memory usage information', 
+          parent=self.find_control('Window', 'Capture Options'))
+      if not collect_system_memory_usage_checkbox.get_toggle_state():
+          collect_system_memory_usage_checkbox.click_input()
+
+  def _close_capture_options_dialog(self):
+      logging.info('Saving "Capture Options"')
+      self.find_control(
+          'Button', 'OK', parent=self.find_control('Window', 'Capture Options')).click_input()
+
+  def _execute(self, memory_sampling_period: str):
+      self._show_capture_options_dialog()
+      self._enable_collect_system_memory_usage()
+      memory_sampling_period_edit = self.find_control('Edit', 'MemorySamplingPeriodEdit')
+      
+      DEFAULT_SAMPLING_PERIOD = "10"
+      expected = ""
+      if not memory_sampling_period:
+          # If the user clean the input, sampling period should be reset to the default value 
+          expected = DEFAULT_SAMPLING_PERIOD
+      elif (not memory_sampling_period.isdecimal()) or (not memory_sampling_period.lstrip('0')):
+          # If the input contains invalid characters (e.g., letters) or its value equals to 0, 
+          # sampling period should keep the same as the previous value.
+          expected = memory_sampling_period_edit.texts()[0]
+      else:
+          expected = memory_sampling_period.lstrip('0')
+
+      logging.info('Entering "%s" in "Sampling period (ms)" edit' % memory_sampling_period)    
```

Comment:
Do we have a policy for python2-style format strings? I personally prefer the python3 ones (`"".format(...)`), but I see the old ones are in use also in other parts of the code. Looping in @reichlfl for the discussion.

Reply:
Not that I know of. The %s formatting is probably my fault to some extent, because I've never gotten used to `.format()` after switching from Python 2 to 3. I don't have a preference.

Reply:
Okay good. Let's just leave it as it is.